torch/custom_class.h

```diff
@@ -18,6 +18,32 @@
 
 namespace torch {
 
+/// This struct is used to represent default values for arguments
+/// when registering methods for custom classes.
+///     static auto register_foo = torch::class_<Foo>("myclasses", "Foo")
+///       .def("myMethod", &Foo::myMethod, {torch::arg("name") = name});
+struct arg {
```

Comment:
this is pretty cool

Reply:
I think the only difference between this and pybind is that in pybind you don't need to pass a list; they use variadic templates to take an arbitrary number of arguments. We *could* do that but it would make the code a little more complex.

Reply:
Could we use variadic arguments as well? Though this is only a minor friction when users migrate from pybind to torchbind, but less friction is still better.

Reply:
I updated the PR to use variadic templates to allow a more pybind-like API but I am not confident in my C++ metaprogramming.

Reply:
It seems we are running into CI errors due to differences between gcc and clang and how they handle parameter packs: https://stackoverflow.com/questions/29098835/can-parameter-pack-function-arguments-be-defaulted